automl/automl-library/library.whizzml

```diff
@@ -352,19 +362,21 @@
         new-fields
         (map (lambda(m) (_batch-association-sets dataset-id m "antecedent" rules))
                         (_model-from-list model-list "association")))
-    (_dataset-with-rules dataset-id name (_flatten new-fields) excluded)))
+    (_dataset-with-rules dataset-id name (flatten new-fields) excluded)))
 
 ;; From a batch score, obtains the generated output dataset
 (define (batch-output-ds res)
   (when (resource-id? res)
     ((fetch (wait res)) "output_dataset_resource" false)))
 
 ;; Creates a new dataset by juxtaposing a list of datasets
-(define (feature-generation-dataset ds-list name objective-id)
-  (update (create-dataset {"origin_datasets" (remove-false ds-list)
-                           "juxtapose" true
-                           "name" (str name " | extended")})
-          {"objective_field" {"id" objective-id}}))
+(define (feature-generation-dataset ds-list name objective-id non-pref)
+  (let (params {"origin_datasets" (remove-false ds-list)
+                "juxtapose" true
+                "name" (str name " | extended")})
+  (set-non-preferred (update (create-dataset params)
```

Comment:
(update ...) returns a boolean, and I think this expects a dataset-id. You can use (update-and-wait ...)